parachain-tracer/src/tracker.rs

```diff
@@ -423,31 +475,36 @@ impl SubxtTracker {
 		metrics: &impl PrometheusMetrics,
 		storage: &TrackerStorage,
 	) {
-		if self.current_candidate.is_idle() {
-			progress.events.push(ParachainConsensusEvent::SkippedSlot);
-			stats.on_skipped_slot(progress);
-			metrics.on_skipped_slot(progress);
-		}
+		for candidate in self.all_candidates() {
+			if candidate.is_idle() {
+				progress.events.push(ParachainConsensusEvent::SkippedSlot);
+				stats.on_skipped_slot(progress);
+				metrics.on_skipped_slot(progress);
+			}
 
-		if self.current_candidate.is_backed() {
-			if let Some(candidate_hash) = self.current_candidate.candidate_hash {
-				progress.events.push(ParachainConsensusEvent::Backed(candidate_hash));
-				stats.on_backed();
-				metrics.on_backed(self.para_id);
+			if candidate.is_backed() {
```

Comment:
There seem to be so many ifs here, a match would be better if we had an enum for the candidate state.